roachtest: activerecord failed

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/21176284?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/21176284?buildTab=artifacts#/activerecord) on master @ [b7e51edafae801342da07734a4d77b53e29084a3](https://github.com/cockroachdb/cockroach/commits/b7e51edafae801342da07734a4d77b53e29084a3):

Failed with:

```
(orm_helpers.go:225).summarizeFailed: 
Tests run on Cockroach v26.2.0-alpha.1-dev-b7e51edafae801342da07734a4d77b53e29084a3
Tests run against activerecord 8.1.0
8908 Total Tests Run
8905 tests passed
3 tests failed
45 tests skipped
46 tests ignored
0 tests passed unexpectedly
2 tests failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: HotCompatibilityTest#test_cleans_up_after_prepared_statement_failure_in_nested_transactions (unexpected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: HotCompatibilityTest#test_cleans_up_after_prepared_statement_failure_in_a_transaction (unexpected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```
Cluster Node to Ip Mapping:
| Node | Public IP | Private IP |
| --- | --- | --- |
| teamcity-21176284-1772263521-13-n1cpu4-0001 | 34.148.30.39 | 10.142.15.193 |

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>diskCount=0</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=true</code>
 - <code>metamorphicLeases=default</code>
 - <code>metamorphicWriteBuffering=true</code>
 - <code>runtimeAssertionsBuild=false</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-21176284/activerecord/1772263661189/1772266331474)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #164486 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot T-sql-foundations branch-release-25.2 release-blocker]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-60880

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/21178832?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/21178832?buildTab=artifacts#/activerecord) on master @ [6807c6539c8803d2c2b2fcf7a66e7ec551d07da0](https://github.com/cockroachdb/cockroach/commits/6807c6539c8803d2c2b2fcf7a66e7ec551d07da0):

Failed with:

```
(orm_helpers.go:225).summarizeFailed: 
Tests run on Cockroach v26.2.0-alpha.1-dev-6807c6539c8803d2c2b2fcf7a66e7ec551d07da0
Tests run against activerecord 8.1.0
8908 Total Tests Run
8905 tests passed
3 tests failed
45 tests skipped
46 tests ignored
0 tests passed unexpectedly
2 tests failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- FAIL: HotCompatibilityTest#test_cleans_up_after_prepared_statement_failure_in_a_transaction (unexpected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: HotCompatibilityTest#test_cleans_up_after_prepared_statement_failure_in_nested_transactions (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/cpu_arch=arm64/run_1
```
Cluster Node to Ip Mapping:
| Node | Public IP | Private IP |
| --- | --- | --- |
| teamcity-21178832-1772436340-09-n1cpu4-0001 | 34.67.150.159 | 10.128.0.143 |

Parameters:
 - <code>arch=arm64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>diskCount=0</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=false</code>
 - <code>metamorphicLeases=default</code>
 - <code>metamorphicWriteBuffering=true</code>
 - <code>runtimeAssertionsBuild=false</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-21178832/activerecord/1772436474869/1772439847255)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #164486 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot T-sql-foundations branch-release-25.2 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/cmd/roachtest/tests/activerecord_blocklist.go
+++ b/pkg/cmd/roachtest/tests/activerecord_blocklist.go
@@ -58,6 +58,8 @@ var activeRecordIgnoreList = blocklist{
 	`HasManyAssociationsTest#test_dependent_association_respects_optional_hash_conditions_on_delete`:                                                           "flaky",
 	`HasManyAssociationsTest#test_dependent_association_respects_optional_sanitized_conditions_on_delete`:                                                      "flaky",
 	`HasManyAssociationsTest#test_depends_and_nullify`:                                                                                                         "flaky",
+	`HotCompatibilityTest#test_cleans_up_after_prepared_statement_failure_in_nested_transactions`:                                                              "flaky",
+	`HotCompatibilityTest#test_cleans_up_after_prepared_statement_failure_in_a_transaction`:                                                                    "flaky",
 	`InheritanceTest#test_destroy_all_within_inheritance`:                                                                                                      "flaky",
 	`InheritanceTest#test_eager_load_belongs_to_primary_key_quoting`:                                                                                           "flaky",
 	`InheritanceTest#test_eager_load_belongs_to_something_inherited`:                                                                                           "flaky",

